src/main/java/org/reactome/release/cosmicupdate/COSMICUpdateUtil.java

```diff
@@ -0,0 +1,320 @@
+		}
+		return updates;
+	}
+
+	/**
+	 * Checks EWASes to see if they have modifiedResidues that have a referenceSequence that is NOT the same as the EWASes referenceEntity.
+	 * The suggested prefix will be set to COSF on the update record if mismatches are found, otherwise COSM will be set.
+	 * @param nonEWASPrinter CSVPrinter for reporting.
+	 * @param identifier Identifier of the object being checked, used for reporting.
+	 * @param updater A COSMICIdentifierUpdater whose suggested prefix will be updated.
+	 * @param EWASes The EWASes to check.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 * @throws IOException
+	 */
+	private static void checkEWASes(CSVPrinter nonEWASPrinter, String identifier, COSMICIdentifierUpdater updater, Collection<GKInstance> EWASes) throws InvalidAttributeException, Exception, IOException
+	{
+		String prefix;
+		for (GKInstance ewas : EWASes)
+		{
+			if (ewas.getSchemClass().getName().equals(ReactomeJavaConstants.EntityWithAccessionedSequence))
+			{
+				GKInstance refSequence = (GKInstance) ewas.getAttributeValue(ReactomeJavaConstants.referenceEntity);
+				// get hasModifiedResidue
+				@SuppressWarnings("unchecked")
+				List<GKInstance> modResidues = (List<GKInstance>) ewas.getAttributeValuesList(ReactomeJavaConstants.hasModifiedResidue);
+				
+				boolean foundMismatchedRefSequence = referenceSequenceMismatchesResidues(refSequence, modResidues);
+				
+				// If we get to the end of the loop and there is a mismatch, then COSF.
+				prefix = foundMismatchedRefSequence ? COSMIC_FUSION_PREFIX : COSMIC_LEGACY_PREFIX;
+				updater.setSuggestedPrefix(prefix);
+			}
+			else
+			{
+				nonEWASPrinter.printRecord(identifier, ewas.toString());
+			}
+		}
+	}
+
+	/**
+	 * Checks modifiedResidues (only FragmentReplacedModification and FragmentInsertionModification are of interest) to see if they match
+	 * refSequence. 
+	 * @param refSequence A Reference Sequence
+	 * @param modResidues The modified residues.
+	 * @return TRUE if there is a mismatch: a mismatch is when the reference sequence DBID != the modified residues's reference sequence's DBID. FALSE, otherwise.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 */
+	private static boolean referenceSequenceMismatchesResidues(GKInstance refSequence, List<GKInstance> modResidues) throws InvalidAttributeException, Exception
+	{
+		boolean foundMismatchedRefSequence = false;
+		int i = 0;
+		while (!foundMismatchedRefSequence && i < modResidues.size())
+		{
+			GKInstance modResidue = modResidues.get(i);
+			if (modResidue.getSchemClass().getName().contains(ReactomeJavaConstants.FragmentReplacedModification)
+				|| modResidue.getSchemClass().getName().contains(ReactomeJavaConstants.FragmentInsertionModification))
+			{
+				GKInstance residueRefSequence = (GKInstance) modResidue.getAttributeValue(ReactomeJavaConstants.referenceSequence);
+				foundMismatchedRefSequence = !residueRefSequence.getDBID().equals(refSequence.getDBID());
```

Comment:
I think I find this line a little tricky to read because its a boolean value that is based on a ! - looking at the code, it doesn't really make sense to re-organize everything, but perhaps a helper method for legibility?

Reply:
I'm not sure what the problem is.

Reply:
I was just saying that its a boolean value created from a !, so I don't know, I guess like saying `boolean colourIsBlue = !object.getColour().isRed()`-- nothing wrong with it, just that reading it took a moment for me to unpack how the statement worked

Reply:
Ok. I don't think there's any further way to simplify this, so I'm going to leave it as it is.